### PR TITLE
removes OIDC auth cookies from request sent to the backend

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- removes OIDC auth cookies from request sent to the backend

## Why are we making these changes?

The backend is not expected to process OIDC auth cookies generated by Waiter, we should avoid sending them to the backend.

### CODE REVIEW

File: waiter/src/waiter/auth/authentication.clj [hunk 1]
```diff
@@ -29,8 +29,14 @@
 
 (def ^:const AUTH-COOKIE-EXPIRES-AT "x-auth-expires-at")
 
+(def ^:const AUTH-COOKIE-EXPIRES-AT-ENTRY-PREFIX (str AUTH-COOKIE-EXPIRES-AT "="))
+
 (def ^:const AUTH-COOKIE-NAME "x-waiter-auth")
 
+(def ^:const AUTH-COOKIE-NAME-ENTRY-PREFIX (str AUTH-COOKIE-NAME "="))
+
+(def ^:const OIDC-CHALLENGE-COOKIE-PREFIX "x-waiter-oidc-challenge-")
```
Comment (shamsimam) [inline waiter/src/waiter/auth/authentication.clj:38]: Unfortunately, we have to share the OIDC cookie name in this file but couldn't come up with a better alternative that would avoid additional computation.

File: waiter/test/waiter/auth/oidc_test.clj [hunk 2]
```diff
@@ -696,3 +696,33 @@
...
+           {:cookie-string "a=b; x-waiter-oidc-challenge-a1b2c3=123; x-waiter-oidc-challenge-3cb2a1=321; c=d" :expected "a=b; c=d"}
+           {:cookie-string "a=b; x-waiter-oidc-challenge-a1b2c3=123; c=d; x-waiter-oidc-challenge-3cb2a1=321" :expected "a=b; c=d"}
+           {:cookie-string "a=b; c=d; x-waiter-oidc-challenge-a1b2c3=123; x-waiter-oidc-challenge-3cb2a1=321" :expected "a=b; c=d"}
+           {:cookie-string "x-waiter-oidc-challenge-a1b2c3=123; x-waiter-oidc-challenge-4d5e6f=456; x-waiter-oidc-challenge-3cb2a1=321; a=b; c=d" :expected "a=b; c=d"}
+           {:cookie-string "x-waiter-oidc-challenge-a1b2c3=123; a=b; x-waiter-oidc-challenge-3cb2a1=321; x-waiter-oidc-challenge-4d5e6f=456; c=d" :expected "a=b; c=d"}
+           {:cookie-string "x-waiter-oidc-challenge-a1b2c3=123; a=b; c=d; x-waiter-oidc-challenge-3cb2a1=321; x-waiter-oidc-challenge-4d5e6f=456" :expected "a=b; c=d"}
+           {:cookie-string "a=b; x-waiter-oidc-challenge-a1b2c3=123; x-waiter-oidc-challenge-3cb2a1=321; c=d; x-waiter-oidc-challenge-4d5e6f=456" :expected "a=b; c=d"}]]
+    (is (= expected (auth/remove-auth-cookie cookie-string)) (str test-case))))
```
Comment (napple) [inline waiter/test/waiter/auth/oidc_test.clj:728]: I don’t see existing cases for `remove-auth-cookie` that tea removal of the other prefixes. Maybe that’s tested indirectly? Should they be tested explicitly like these oidc cases?

I might also include cases where the prefix string is present but it’s not used as a prefix, eg `blah-x-waiter-oidc-challenge-abc=123` should not be removed.

Comment (napple) [reply waiter/test/waiter/auth/oidc_test.clj:728]: Maybe include a case with an empty cookie string, too.

File: waiter/test/waiter/auth/oidc_test.clj [hunk 3]
```diff
@@ -696,3 +696,33 @@
...
+(deftest test-remove-challenge-cookies
+  (doseq [{:keys [cookie-string expected] :as test-case}
+          [{:cookie-string nil :expected nil}
+           {:cookie-string "a=b" :expected "a=b"}
+           {:cookie-string "a=b; c=d" :expected "a=b; c=d"}
+           {:cookie-string "x-waiter-oidc-challenge-A1B2C3=1234; a=b" :expected "a=b"}
+           {:cookie-string "a=b; x-waiter-oidc-challenge-A1B2C3=1234" :expected "a=b"}
+           {:cookie-string "x-waiter-oidc-challenge-a1b2c3=1234; a=b" :expected "a=b"}
+           {:cookie-string "a=b; x-waiter-oidc-challenge-a1b2c3=1234" :expected "a=b"}
+           {:cookie-string "a=b; x-waiter-oidc-challenge-a1b2c3=1234" :expected "a=b"}
```
[the remark this answers is not in the document]
Comment (shamsimam) [reply waiter/test/waiter/auth/oidc_test.clj]: Removed.

File: waiter/test/waiter/auth/oidc_test.clj (hunk 2 above)
Comment (shamsimam) [reply waiter/test/waiter/auth/oidc_test.clj:730]: Added additional test cases and a separate unit test function.

**Review by napple — APPROVED** [review]

**Pull request merged**